src/blackdoor/cqbe/node/Node.java

```diff
@@ -174,30 +205,49 @@ public void setBootstrapNode(L3Address bootstrapNode) {
 			this.bootstrapNode = bootstrapNode;
 		}
 
+		public void setLocal(boolean local) {
+			this.local = local;
+
+		}
+
 		/**
 		 * Builds a node based on the current list of settings attributed to it.
 		 * TODO add and start updater
+		 * 
+		 * @throws Exception
 		 */
-		public Node buildNode() throws RequiredParametersNotSetException {
-			Node node = null;
-			if (!daemon) {
-				if (adam) {
-					AddressTable builderAddressTable = new AddressTable();
-					node = new Node(port);
-					node.addressTable = builderAddressTable;
-				} else if (!adam && bootstrapNode != null) {
-					AddressTable builderAddressTable = new AddressTable();
-					builderAddressTable.add(bootstrapNode);
-					node = new Node(port);
-					node.addressTable = builderAddressTable;
-				} else
-					throw new RequiredParametersNotSetException();
-			} else {
-				// TODO start new process
+		public Node buildNode() throws Exception {
+			config.saveSessionToFile();
+			if (daemon) {
+				// TODO start a demon prossess depending on platform
+			}
+			Node node = new Node(port);
+			node.configureAddressing(port, local);
+			node.storageController = new StorageController(new File(
+					this.storageDir).toPath(), node.addressTable);
+			if (!adam && bootstrapNode != null) {
+				node.addressTable.add(bootstrapNode);
```

Comment:
we need to either work with @euphzji to explicitly call updater (or make a TODO comment and open an issue indicating our need to do so in the future) or start to populate the address table ourselves when the node is created.